File: src/test/java/mnm/mods/util/test/LocationTests.java
package mnm.mods.util.test;

import static org.junit.Assert.*;

import org.junit.Test;

import mnm.mods.util.ILocation;
import mnm.mods.util.Location;

public class LocationTests {

    @Test
    public void contains() {

        // inside
        ILocation loc1 = new Location(1, 1, 16, 16);
        ILocation loc2 = new Location(2, 2, 15, 15);
        assertTrue(loc1.contains(loc2));
        assertTrue(loc2.contains(loc1));

        // corner intersection
        loc1 = new Location(15, 0, 12, 6);
        loc2 = new Location(17, 4, 12, 6);
        assertTrue(loc1.contains(loc2));
        assertTrue(loc2.contains(loc1));

        // sides
        loc1 = new Location(16, 16, 16, 16);
        loc2 = new Location(18, 16, 16, 16);
        assertTrue(loc1.contains(loc2));
        assertTrue(loc2.contains(loc1));

        loc1 = new Location(5, 5, 10, 2);
        loc2 = new Location(16, 9, 1, 1);
        assertFalse(loc1.contains(loc2));
        assertFalse(loc2.contains(loc1));

        // edge cases
        loc1 = new Location(2, 1, 1, 1);
        loc2 = new Location(1, 1, 1, 1);
        assertFalse(loc1.contains(loc2));
        assertFalse(loc2.contains(loc1));

        loc1 = new Location(16, 16, 16, 16);
        loc2 = new Location(16, 32, 16, 16);
        assertFalse(loc1.contains(loc2));
        assertFalse(loc2.contains(loc1));
    }

}

File: src/test/kotlin/mnm/mods/util/test/LocationTests.kt
package mnm.mods.util.test

import org.junit.Assert.*

import org.junit.Test

import mnm.mods.tabbychat.util.ILocation
import mnm.mods.tabbychat.util.Location

class LocationTests {

    @Test
    fun contains() {

        // inside
        var loc1: ILocation = Location(1, 1, 16, 16)
        var loc2: ILocation = Location(2, 2, 15, 15)
        assertTrue(loc2 in loc1)
        assertTrue(loc1 in loc2)

        // corner intersection
        loc1 = Location(15, 0, 12, 6)
        loc2 = Location(17, 4, 12, 6)
        assertTrue(loc2 in loc1)
        assertTrue(loc1 in loc2)

        // sides
        loc1 = Location(16, 16, 16, 16)
        loc2 = Location(18, 16, 16, 16)
        assertTrue(loc2 in loc1)
        assertTrue(loc1 in loc2)

        loc1 = Location(5, 5, 10, 2)
        loc2 = Location(16, 9, 1, 1)
        assertFalse(loc2 in loc1)
        assertFalse(loc1 in loc2)

        // edge cases
        loc1 = Location(2, 1, 1, 1)
        loc2 = Location(1, 1, 1, 1)
        assertFalse(loc2 in loc1)
        assertFalse(loc1 in loc2)

        loc1 = Location(16, 16, 16, 16)
        loc2 = Location(16, 32, 16, 16)
        assertFalse(loc2 in loc1)
        assertFalse(loc1 in loc2)
    }

}
